security-service-client-spring/src/main/java/org/zowe/apiml/security/client/service/GatewaySecurityService.java

```diff
@@ -70,6 +65,8 @@ public Optional<String> login(String username, String password) {
             ErrorType errorType = getErrorType(e);
             responseHandler.handleBadResponse(e, errorType,
                 "Cannot access Gateway service. Uri '{}' returned: {}", uri, e.getMessage());
+        } finally {
+            loginRequest.evictSensitiveData();
```

Comment:
Is this really the correct place? The array in the argument will be erased. I guess it is the responsibility of the caller.